mico-core/src/main/java/io/github/ust/mico/core/service/MicoStatusService.java

```diff
@@ -33,95 +57,151 @@
+     * (cpu load, memory load)
+     *
      * @param micoService is a {@link MicoService}
      * @return {@link MicoServiceStatusDTO} which contains status information for a specific {@link MicoService}
      */
     public MicoServiceStatusDTO getServiceStatus(MicoService micoService) {
-        Optional<Deployment> deploymentOptional = null;
+        MicoServiceStatusDTO serviceStatus = new MicoServiceStatusDTO();
         try {
-            deploymentOptional = micoKubernetesClient.getDeploymentOfMicoService(micoService);
+            Optional<Deployment> deploymentOptional = micoKubernetesClient.getDeploymentOfMicoService(micoService);
+            if (deploymentOptional.isPresent()) {
+                Deployment deployment = deploymentOptional.get();
+                serviceStatus.setAvailableReplicas(deployment.getStatus().getAvailableReplicas());
+                serviceStatus.setRequestedReplicas(deployment.getSpec().getReplicas());
+            } else {
+                log.warn("There is no deployment of the MicoService '{}' '{}'. Continue with next one.",
+                    micoService.getShortName(), micoService.getVersion());
+                return serviceStatus.setErrorMessages(CollectionUtils.listOf("No deployment of " + micoService.getShortName() + " " + micoService.getVersion() + " is available."));
+            }
         } catch (KubernetesResourceException e) {
             log.error("Error while retrieving Kubernetes deployment of MicoService '{}' '{}'. Continue with next one. Caused by: {}",
                 micoService.getShortName(), micoService.getVersion(), e.getMessage());
+            return serviceStatus.setErrorMessages(CollectionUtils.listOf("Error while retrieving Kubernetes deployment of MicoService " + micoService.getShortName() + " "
+                + micoService.getVersion() + " .  Caused by: " + e.getMessage()));
         }
-        if (!deploymentOptional.isPresent()) {
-            log.warn("There is no deployment of the MicoService '{}' '{}'. Continue with next one.",
-                micoService.getShortName(), micoService.getVersion());
-        }
+        serviceStatus
+            .setName(micoService.getName())
+            .setShortName(micoService.getShortName())
+            .setVersion(micoService.getVersion());
 
-        Deployment deployment = deploymentOptional.get();
-        MicoServiceStatusDTO serviceStatus = new MicoServiceStatusDTO();
-        serviceStatus.setName(micoService.getName());
-        serviceStatus.setShortName(micoService.getShortName());
-        serviceStatus.setVersion(micoService.getVersion());
-        serviceStatus.setAvailableReplicas(deployment.getStatus().getAvailableReplicas());
-        serviceStatus.setRequestedReplicas(deployment.getSpec().getReplicas());
+        // Get status information for service interfaces of a service
+        serviceStatus.setInterfacesInformation(getServiceInterfaceStatus(micoService));
```

Comment:
`getServiceInterface` may throw an exception. If so, catch it and add the error to the error message list.

Reply:
Also consider to improve the overall error handling of `getServiceInterfaceStatus`

Reply:
What about if we set a booelan **kubernetesServiceExists** in each MicoServiceInterfaceStatusDTO?

Reply:
Like discussed [here](https://github.com/UST-MICO/mico/pull/485/files#r260858203) the *MicoServiceInterfaceStatusDTO* will already have a property `publicIPs`. If this list contains at least one IP address, there is a Kubernetes Service.
Therefore I think it is not necessary to also add a boolean `kubernetesServiceExists`.

Reply:
I will add a check for an empty list of `publicIps` in getServiceStatus() and add a message entry to `errorMessages` if it is empty.